AsyncDisplayKit/ASVideoNode.h

```diff
@@ -0,0 +1,34 @@
+
+#import <AsyncDisplayKit/AsyncDisplayKit.h>
+
+typedef NS_ENUM(NSUInteger, ASVideoGravity) {
+  ASVideoGravityResizeAspect,
+  ASVideoGravityResizeAspectFill,
+  ASVideoGravityResize
+};
+
+@protocol ASVideoNodeDelegate;
+
+@interface ASVideoNode : ASControlNode<_ASDisplayLayerDelegate>
+@property (atomic, strong, readwrite) AVAsset *asset;
```

Comment:
Note - although the atomic specifier is a good self-documentation hint to developers that these methods are thread safe, make sure not to use it if you are going to let the compiler synthesize either the getter or the setter for you.  It creates very slow locking that we should avoid.  So, best option is to implement all the getters / setters manually and grab the MutexLocker yourself, like ASDisplayNode.